Documentation/upgrading-from-ipy2.md

```diff
@@ -14,3 +14,167 @@ In an effort to improve compatibility, `sys.platform` no longer returns `cli`. I
+>>> hasattr(j, 'MaxValue') 
+False
+>>> hex(j)
+'0x80000000'
+>>> j = int(j)  # no type change, j stays BigInteger
+>>> hasattr(j, 'MaxValue')
+False
+>>> j = System.Int32(j)  # conversion fails
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+OverflowError: Arithmetic operation resulted in an overflow.
+```
+
+Such explicit conversions are in most cases unnecessary since the runtime recognizes `int`/`Int32` equivalence of instances and performs necessary conversions automatically.
+
+```pycon
+>>> import System
+>>> int_list = System.Collections.Generic.List[int]()
+>>> int_list.Add(1) # Int32 instance converted to BigInteger
+>>> int32_list = System.Collections.Generic.List[System.Int32]()
+>>> int32_list.Add((1).ToBigInteger()) # BigInteger instance converted to Int32
+>>> int_list[0] == int32_list[0]
+True
+```
+
+### Pickling and unpickling of `int`
+
+When an `int` object is serialized using `pickle.dump(x, myfile)` and subsequently unpickled with `x = pickle.load(myfile)` (or `pickle.loads(pickle.dumps(x))`, this has the same effect as reconstructing the object using the `int` constructor, i.e. `x = int(x)`. In other words, if the `x` instance was `BigInteger` but the value fits in `Int32`, it will be reconstructed as `Int32`.
+
+### BigIntegerV2 API
+
+In IronPython 2, `long` type carries an obsolete `BigIntegerV2` API, accessible after importing `System`. In IronPython 3 this API is not available directly on `int` instances (regardless whether the instance is `Int32` or `BigInteger`), but is still accessible in some form through `Microsoft.Scripting.Utils.MathUtils` in `Microsoft.Dynamic.dll`.
+
+```pycon
+>>> # IronPython 2
+>>> i = 1        # instance of Int32 (int)
+>>> j = 1 << 64  # instance of BigInteger (long)
+>>> import System
+>>> j.GetWords()
+Array[UInt32]((0, 0, 1))
+>>> i.GetWords()
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+AttributeError: 'int' object has no attribute 'GetWords'
+>>> long.GetWords(i)
+Array[UInt32]((1))
+```
+
+```pycon
+>>> # IronPython 3
+>>> i = 1        # instance of Int32 (int)
+>>> j = 1 << 64  # instance of BigInteger (int)
+>>> import clr
+>>> clr.AddReference("Microsoft.Dynamic")
+>>> import Microsoft.Scripting.Utils.MathUtils
+>>> clr.ImportExtensions(Microsoft.Scripting.Utils.MathUtils)
+>>> j.GetWords()
+Array[UInt32]((0, 0, 1))
+>>> i.GetWords()
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+AttributeError: 'int' object has no attribute 'GetWords'
+>>> Microsoft.Scripting.Utils.MathUtils.GetWords(i)
+Array[UInt32]((1))
+```
+
+Another set of Python-hidden methods on `long` in IronPython 2 that not available on `int` in IronPython 3 are conversion methods with names like `ToXxx`. The recommended way to perform type conversions like those is to use type constructors. The exception is the conversion to `BigInteger` itself, for the reasons explained above.
```

Comment:
typo: missing "are" in "that not available"

I thought you had added all the `IConvertible` like methods to `int`? Though I guess those all take an `IFormatProvider` as an argument...

Reply:
Yes, they are different `ToXxx` methods. Keeping them on `BigInteger` seemed superfluous since the job can be easily done using relevant constructors (which feels more Pythonic) and also if they stayed in `BigInteger` they would have to be added on `int32` too. Looked just like pollution.